lib/juttle-service-client.js

```diff
@@ -0,0 +1,290 @@
+    console.log('                                              Used by "run", "get_inputs", "push", "watch".');
+    console.log('       --wait                                 If true, wait for program to finish');
+    console.log('                                                (default false, job starts in background)');
+    console.log('       --input name=val:                      One or more input values.');
+    console.log('                                              Used by "get_inputs".');
+    console.log('       --job <job-id>:                        Job id.');
+    console.log('                                              Used by "subscribe", "list_jobs", "delete".');
+    console.log('       --observer <observer-id>:              Observer id.');
+    console.log('                                              Used by "subscribe", "run"');
+    console.log('       --juttle-service <hostname:port>:       Hostname/port of juttle-service juttle server');
+    for (let option of addl_options) {
+        console.log('         ' + option);
+    }
+    console.log('       --help:                                Print this help and exit');
+    process.exit(1);
+}
+
+function handle_error(err, filename) {
+
+    // The error returned when the request-promise module rejects a
+    // promise due to a non-200 status contains an error property that
+    // contains the body of the response. (Other errors may not
+    // contain this property). If this exists and is JSON, parse it
+    // and look for a location. Otherwise, just print the error
+    // message.
+
+    let err_obj;
+    try {
+        err_obj = JSON.parse(err.error);
+    } catch (e) {
+        // err did not contain a error property or it was not
+        // json.
+        console.error('ERROR', err.message);
+        return;
+    }
+
+    // If it's a juttle error containing a program
+    // location, print the error in context.
+    if (err_obj.code === 'JS-JUTTLE-ERROR') {
+        console.error(cli_errors.show_in_context({err: err_obj.info.err,
+                                                  program: err_obj.info.bundle.program,
+                                                  modules: err_obj.info.bundle.modules,
+                                                  filename: filename}));
+    } else {
+        console.error('ERROR', err.message);
+    }
+}
+
+function read_inputs(opts) {
+
+    let inputs = {};
+    if (opts.input !== undefined) {
+        if (! _.isArray(opts.input)) {
+            opts.input = [opts.input];
+        }
+        _.each(opts.input, function(x) {
+            let L = x.split('=');
+            if (L.length !== 2) {
+                console.error('error: invalid input', x);
+                usage();
+            }
+            let num = +L[1];
+            inputs[L[0]] = isNaN(num) ? L[1] : num;
+        });
+    }
+
+    return inputs;
+}
+
+function handle_command(opts, command) {
```

Comment:
Sure, but as you mentioned before won't the future `juttle-engine-client` have additional commands that it will implement?

Also FWIW I could see using the exported command functions programmatically in other contexts, so even if you end up keeping handle_command as part of the lib module, I could see value in breaking out the body of the switch into individually exported functions.